feat: add security headers to HTTP server for enhanced protection

BenElferink opened the pull request:
This commit introduces security headers including Content-Security-Policy, Strict-Transport-Security, and X-Frame-Options to the HTTP server, improving the overall security posture of the application.

emergency-ticket id: EMR-7

BenElferink committed:
feat: add security headers to HTTP server for enhanced protection

This commit introduces security headers including Content-Security-Policy, Strict-Transport-Security, and X-Frame-Options to the HTTP server, improving the overall security posture of the application.

BenElferink committed:
Merge branch 'main' into ben/plat-190-fixing-missing-security-headers-csp-hsts

alonkeyval commented on frontend/main.go:
```diff
@@ -145,6 +145,14 @@ func startHTTPServer(ctx context.Context, flags *Flags, logger logr.Logger, odig
 	r.Use(cors.Default())
 
+	// Enable security headers
+	r.Use(func(c *gin.Context) {
```
nit: if it works correctly and does not break the ui as the cursor bot pointed out, I would move the headers to a func and use it:


```suggestion
// Add security headers middleware
r.Use(securityHeadersMiddleware())
```

BenElferink committed:
fix: update Content-Security-Policy to allow inline scripts and styles

This commit modifies the Content-Security-Policy header to permit 'unsafe-inline' for both scripts and styles, enhancing flexibility while maintaining security measures.

BenElferink committed:
chore: bump deps

BenElferink committed:
Merge branch 'main' into ben/plat-190-fixing-missing-security-headers-csp-hsts

BenElferink committed:
refactor: move security headers to middleware for better organization

This commit refactors the security headers implementation by moving it to a dedicated middleware function, enhancing code organization and maintainability. The middleware sets the Content-Security-Policy, Strict-Transport-Security, and X-Frame-Options headers for improved security.

BenElferink committed:
Merge branch 'ben/plat-190-fixing-missing-security-headers-csp-hsts' of github.com:BenElferink/odigos into ben/plat-190-fixing-missing-security-headers-csp-hsts

BenElferink committed:
fix: fonts
